lib/mayaUsd/fileio/primUpdater.cpp

```diff
@@ -16,30 +16,191 @@
+        return SdfShouldCopyValue(
+            topSrcPath,
+            topDstPath,
+            specType,
+            field,
+            srcLayer,
+            srcPath,
+            fieldInSrc,
+            dstLayer,
+            dstPath,
+            fieldInDst,
+            valueToCopy);
+    };
+    auto dontCopyChildrenFn = [](const TfToken& childrenField,
+                                 const SdfLayerHandle&,
+                                 const SdfPath&,
+                                 bool,
+                                 const SdfLayerHandle&,
+                                 const SdfPath&,
+                                 bool,
+                                 boost::optional<VtValue>*,
+                                 boost::optional<VtValue>*) {
+        // There must be an existing list of static children fields.
+        // PPT, 18-Oct-21.
+        static TfToken properties("properties");
+        static TfToken primChildren("primChildren");
 
-bool UsdMayaPrimUpdater::Pull(UsdMayaPrimUpdaterContext* context) { return false; }
+        // See traverseLayer() implementation for full list of children field.
+        // Property children must be copied, prim children must not.  What
+        // about other children field?  If we understand the complete list, we
+        // can encode it in a map, rather than a chain of conditionals.  PPT,
+        // 18-Oct-21.
+        if (childrenField == properties) {
+            return true;
+        } else if (childrenField == primChildren) {
+            return false;
+        }
 
-void UsdMayaPrimUpdater::Clear(UsdMayaPrimUpdaterContext* context) { }
+        // Default is copy.
+        return true;
+    };
+
+    return SdfCopySpec(
+        srcLayer, topSrcPath, dstLayer, topDstPath, shouldCopyValueFn, dontCopyChildrenFn);
+}
 
-const MDagPath& UsdMayaPrimUpdater::GetDagPath() const { return _dagPath; }
+const MObject& UsdMayaPrimUpdater::getMayaObject() const { return _mayaObject; }
 
-const MObject& UsdMayaPrimUpdater::GetMayaObject() const { return _mayaObject; }
+const Ufe::Path& UsdMayaPrimUpdater::getUfePath() const { return _path; }
+
+UsdPrim UsdMayaPrimUpdater::getUsdPrim(const UsdMayaPrimUpdaterContext& context) const
+{
+    return MayaUsd::ufe::ufePathToPrim(_path);
+}
 
-const SdfPath& UsdMayaPrimUpdater::GetUsdPath() const { return _usdPath; }
+/* static */
+bool UsdMayaPrimUpdater::isAnimated(const MDagPath& path)
+{
+    auto isDagPathAnimated = [](const MDagPath& dagPath) {
+        int     upstreamDependencies = -1;
+        MString pyCommand;
+        pyCommand.format(
+            "import maya.cmds as cmds\n"
+            "if cmds.evaluationManager( query=True, invalidate=True ):\n"
+            "  upstream = cmds.evaluationManager(ust='^1s')\n"
+            "  for node in upstream:\n"
+            "    if cmds.nodeType(node) == 'mayaUsdProxyShape':\n"
+            "      countNonProxyShape -= 1\n"
```

Comment:
I don't see where countNonProxyShape gets initialized so 1 can be subtracted from it.

Reply:
Absolutely, this will generate a NameError exception.  Will investigate, good catch.